frontend/src/scenes/feature-flags/featureFlagLogic.ts

```diff
@@ -404,7 +404,21 @@ export const featureFlagLogic = kea<featureFlagLogicType>([
                     if (!state) {
                         return state
                     }
-                    return { ...state, filters: { ...state.filters, multivariate: multivariateOptions } }
+                    const variantsSet = new Set(multivariateOptions?.variants.map((variant) => variant.key))
+                    const groups = state.filters.groups.map((group) =>
+                        !group.variant || variantsSet.has(group.variant) ? group : { ...group, variant: null }
+                    )
+                    const oldPayloads = state.filters.payloads ?? {}
+                    const payloads = Object.keys(oldPayloads).reduce((newPayloads, variantKey) => {
+                        if (variantsSet.has(variantKey)) {
+                            return { ...newPayloads, [variantKey]: oldPayloads[variantKey] }
+                        }
+                        return newPayloads
+                    }, {})
+                    return {
+                        ...state,
+                        filters: { ...state.filters, groups, payloads, multivariate: multivariateOptions },
+                    }
```

Comment:
This seems more complicated than it should be, considering that this action is only called so far with either `null` or `EMPTY_MULTIVARIATE_OPTIONS` 
https://github.com/PostHog/posthog/blob/e2bdc9bd8f52c89fd57f310039a13b984c115066/frontend/src/scenes/feature-flags/featureFlagLogic.ts#L1017-L1021

However, the name of the action implies that a "non-empty" multivariate definition could be set. I think we could simplify this by splitting it into two actions, one that `clears` all the variants (and related data), and the other that initializes the variants for the first time (which happens when the "Served Value" option is selected). I don't have any strong opinions since the current code could handle more scenarios in the future, but please let me know what you think.